### ISSUE TITLE
USWDS - Date picker: Manually formatting date input is difficult

### ISSUE THREAD

**Issue by mahoneycm** [uswds/uswds#5534]
# Summary
Users found it challenging to manually type dates into the date picker input element. 

## Observations
1. It is not always obvious to screen reader users what the accepted date format is.
1. It is easy to make errors when trying to type slashes, especially for blind users. Some users prefer the slashes to be entered automatically.
1. It is not obvious when users make an error in manually typing the date due to a lack of error feedback. 

## Affected user groups
- All users
- Screen reader users
- Screen magnification users

## Research method
1-hour 20 minute moderated usability testing sessions conducted via video conference. Conducted with 5 participants using assistive technology (screen readers and screen magnification) on a high fidelity form prototype.

More details in the [findings report](https://docs.google.com/document/d/1z1BRxYZq_m65IdsRKsMzDYSpYYHKsqYgelQe7w_qDUQ/edit) (Google docs :lock:).

## Recommendations for next steps
1. Provide helpful error feedback when incorrect format is entered. 
1. Explicitly call out in our guidance that the component should always allow for the date to also be typed in manually.
1. Consider adding [input mask](https://designsystem.digital.gov/components/input-mask/) functionality.
1. Implement conditional formatting:
    1. Automatically add `0` for single-digit dates.
    1. Automatically enter slashes so the user does not have to type them.

**Comment by mejiaj** [comment]
@jaclinec, @mahoneycm this looks ready to start, we just need help with writing up the guidance. If there's additional work required, please outline the tasks in the issue description.

CC @sarah-sch

**Comment by mejiaj** [comment]
@mahoneycm @jaclinec 

## Component Feedback

Can you provide an example message or use case?
> - [ ] Provide helpful error feedback when incorrect format is entered.

## Masking

Could you add a little more information on what functionality exactly we need from input mask? Is it the underscores?

>- [ ] Consider adding input mask functionality.

**Comment by jaclinec** [comment]
@mejiaj the [W3C design system offers some examples of form errors for dates ](https://design-system.w3.org/styles/form-errors.html)that look promising:
<img width="530" alt="image" src="https://github.com/uswds/uswds/assets/109540526/b851405c-1996-442a-b21b-9ae95b0eb9d8">

We could try something similar to this, but if the user enters the wrong format the error message could say something like "Please enter a valid date format MM/DD/YYYY" or similar.

I'll add this comment to #5692 as well.

**Comment by mahoneycm** [comment]
@jaclinec I agree!

I created https://github.com/uswds/uswds-site/issues/2476 in hopes of adding guidance to users on how and when to use these styles.

I also noticed that we currently have styles written to remove that sidebar in desktop widths. I think it'd be helpful to users if that error bar showed in all widths to visually group error states. 

Example:

[Kapture 2024-03-04 at 09.36.10.webm](https://github.com/uswds/uswds/assets/25211650/23ee2f5f-4924-4f18-be53-b071842fdc18)

Should we add this error state to our next round of user testing for feedback?

**Comment by jaclinec** [comment]
@mahoneycm totally agree that the error bar should show in all widths if possible. Otherwise, we are relying heavily on the red color to depict that there's an error which isn't accessible for color blind users.

And I'm glad you're thinking about error states and how we need to offer more guidance there! I have some thoughts and also want to share some [great research the patterns team did](https://docs.google.com/presentation/d/1SmjPMxol6JNpMaUhzXtVvkNcYXmnRsP-y-NOozKNuA0/edit?usp=sharing) around this - I'll add it to [#2476](https://github.com/uswds/uswds-site/issues/2476). But generally, I think yes we need to decide on a fairly consistent look, functionality, and guidance around error states and should test it with users when we feel its in a good place to test!

**Comment by amycole501** [comment]
Checking to see where we are with this and if we can release it prior to the January MC @mahoneycm
